### PR TITLE
correct flag order in make-help

### PR DESCRIPTION
**Pull request by BenTheElder**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: super minor correctness fix for `make help`'s implementation, with GNU's find options need to come before the find expression.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/make-rules/make-help.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ readonly reset=$(tput sgr0)
 
 KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/../..
 ALL_TARGETS=$(make -C "${KUBE_ROOT}" PRINT_HELP=y -rpn | sed -n -e '/^$/ { n ; /^[^ .#][^ ]*:/ { s/:.*$// ; p ; } ; }' | sort)
-CMD_TARGETS=$(cd "${KUBE_ROOT}/cmd"; find . -type d -mindepth 1 -maxdepth 1 | cut -c 3-)
```
Comment (BenTheElder) [inline hack/make-rules/make-help.sh]: I could have sworn I'd already done this. I _have_ done this in hack/lib/util.sh

Comment (BenTheElder) [reply hack/make-rules/make-help.sh]: xref: https://github.com/kubernetes/release/pull/802

**Review by JAORMX — APPROVED** [review]

**Pull request merged**